orc/src/test/java/org/apache/iceberg/orc/TestORCSchemaUtil.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+import org.apache.orc.TypeDescription;
+import org.junit.Test;
+
+import static org.apache.iceberg.AssertHelpers.assertThrows;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.junit.Assert.assertEquals;
+
+public class TestORCSchemaUtil {
+
+  @Test
+  public void testRoundtripConversionPrimitive() {
+    Schema expectedSchema = new Schema(
+        optional(1, "intCol", Types.IntegerType.get()),
+        optional(3, "longCol", Types.LongType.get()),
+        optional(6, "intCol2", Types.IntegerType.get()),
+        optional(20, "intCol3", Types.IntegerType.get()),
+        required(9, "doubleCol", Types.DoubleType.get()),
+        required(10, "uuidCol", Types.UUIDType.get()),
+        optional(2, "booleanCol", Types.BooleanType.get()),
+        optional(21, "fixedCol", Types.FixedType.ofLength(4096)),
+        required(22, "binaryCol", Types.BinaryType.get()),
+        required(23, "stringCol", Types.StringType.get()),
+        required(24, "decimalCol", Types.DecimalType.of(15, 3)),
+        required(25, "floatCol", Types.FloatType.get()),
+        optional(30, "dateCol", Types.DateType.get()),
+        required(32, "timeCol", Types.TimeType.get()),
+        required(34, "timestampCol", Types.TimestampType.withZone())
+    );
+    TypeDescription orcSchema = ORCSchemaUtil.convert(expectedSchema);
+    assertEquals(expectedSchema.asStruct(), ORCSchemaUtil.convert(orcSchema).asStruct());
+  }
+
+  @Test
+  public void testRoundtripConversionNested() {
```

Comment:
I think this should also test round-trip conversion for lists, maps, and a complex nesting case with lists of structs and maps of structs.

Reply:
I've added more complex structures to test.